google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java

```diff
@@ -0,0 +1,218 @@
+    if (!appEngineStandardLibraryPanel.isEnabled()) {
+      return false;
+    }
+
+    Set<AppEngineStandardMavenLibrary> savedLibs
+        = facetConfiguration.getLibraries(context.getProject());
+    Set<AppEngineStandardMavenLibrary> selectedLibs
+        = appEngineStandardLibraryPanel.getSelectedLibraries();
+
+    return !Objects.equals(savedLibs, selectedLibs);
+  }
+
+  @Override
+  public void apply() {
+    if (appEngineStandardLibraryPanel.isEnabled()) {
+      Set<AppEngineStandardMavenLibrary> savedLibs
+          = facetConfiguration.getLibraries(context.getProject());
+      Set<AppEngineStandardMavenLibrary> selectedLibs
+          = appEngineStandardLibraryPanel.getSelectedLibraries();
+
+      Set<AppEngineStandardMavenLibrary> libsToAdd = Sets.difference(selectedLibs, savedLibs);
+      Set<AppEngineStandardMavenLibrary> libsToRemove = Sets.difference(savedLibs, selectedLibs);
+
+      if (!libsToAdd.isEmpty()) {
+        for (AppEngineStandardMavenLibrary library : libsToAdd) {
+          AppEngineSupportProvider.loadMavenLibrary(context.getModule(), library);
+        }
+      }
+
+      if (!libsToRemove.isEmpty()) {
+        AppEngineSupportProvider.removeMavenLibraries(libsToRemove, context.getModule());
+      }
+    }
+  }
+
+  @Override
+  public void reset() {
+    if (appEngineStandardLibraryPanel.isEnabled()) {
+      appEngineStandardLibraryPanel
+          .setSelectLibraries(facetConfiguration.getLibraries(context.getProject()));
+    }
+  }
+
+  @SuppressWarnings("checkstyle:abbreviationaswordinname")
+  public void disposeUIResources() {
+    if (libraryListener != null) {
+      LibraryTablesRegistrar.getInstance()
+          .getLibraryTable(context.getProject()).removeListener(libraryListener);
+    }
+  }
+
+  @Override
+  public String getHelpTopic() {
+    return "Google_App_Engine_Facet";
+  }
+
+  @Override
+  public void onFacetInitialized(@NotNull Facet facet) {
+    AppEngineWebIntegration.getInstance().setupDevServer();
+
+    // Called on explicitly adding the facet through Project Settings -> Facets, but not on the
+    // Framework discovered "Configure" popup.
+    UsageTrackerProvider.getInstance()
+        .trackEvent(GctTracking.APP_ENGINE_ADD_FACET)
+        .withLabel("setOnModule")
+        .ping();
+  }
+
+  @SuppressWarnings("checkstyle:abbreviationaswordinname")
+  private void createUIComponents() {
+    appEngineStandardLibraryPanel = new AppEngineStandardLibraryPanel(isManagedLibrariesEnabled());
+  }
+
+  private boolean isManagedLibrariesEnabled() {
+    return !AppEngineProjectService.getInstance().isMavenModule(context.getModule());
+  }
+
+  public class LibraryModificationListener implements Listener {
+
+    @Override
+    public void afterLibraryAdded(final Library addedLibrary) {
+      final DependencyScope scope = AppEngineStandardMavenLibrary
+          .getLibraryByMavenDisplayName(addedLibrary.getName()).getScope();
+
+      new WriteAction() {
```

Comment:
modifying and committing the root model needs to be in a write action from what I've seen in the platform code. I'll double check this.